x-pack/libbeat/common/aws/credentials.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package aws
+
+import (
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/defaults"
+	"github.com/aws/aws-sdk-go-v2/aws/external"
+)
+
+// ConfigAWS is a structure defined for AWS credentials
+type ConfigAWS struct {
+	AccessKeyID               string   `config:"access_key_id"`
+	SecretAccessKey           string   `config:"secret_access_key"`
+	SessionToken              string   `config:"session_token"`
+	ProfileName               string   `config:"credential_profile_name"`
+}
+
+// GetAWSCredentials function gets aws credentials from the config.
+// If access_key_id and secret_access_key are given, then use them as credentials.
+// If not, then load from aws config file. If credential_profile_name is not
+// given, then load default profile from the aws config file.
+func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
+	// Check if accessKeyID and secretAccessKey is given from configuration
+	if config.AccessKeyID != "" && config.SecretAccessKey != "" {
```

Comment:
what if the user provides only one of these? something would be wrong but probably loading default config wouldn't be the expected behavior.

Reply:
Yes good point! I changed it to checking if user provides one of `accessKeyID`, `secretAccessKey` or `sessionToken`, then use `accessKeyID` and `secretAccessKey` and/or `sessionToken`. If `accessKeyID` or `secretAccessKey` is missing, then api call will fail with invalid aws credentials.